tests/utils.go

```diff
@@ -1694,8 +1695,47 @@ func removeNamespaces() {
 	fmt.Println("")
 	for _, namespace := range testNamespaces {
 		fmt.Printf("Waiting for namespace %s to be removed, this can take a while ...\n", namespace)
-		EventuallyWithOffset(1, func() bool { return errors.IsNotFound(virtCli.CoreV1().Namespaces().Delete(namespace, nil)) }, 240*time.Second, 1*time.Second).
-			Should(BeTrue())
+		EventuallyWithOffset(1, func() bool {
+			namespaceDeletionError := virtCli.CoreV1().Namespaces().Delete(namespace, nil)
+			if namespaceDeletionError != nil {
+				fmt.Printf("could not get delete '%v':\n %v\n", namespace, namespaceDeletionError)
+			}
+
+			namespaceObject, err := virtCli.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+			if err != nil {
+				fmt.Printf("could not get namespace '%v':\n %v\n", namespace, err)
+			}
+
+			fmt.Printf("deleting namespace '%v':\n %v\n", namespace, err)
+			fmt.Printf("finalizers %v \n", namespaceObject.Finalizers)
+			fmt.Printf("OwnerReferences %v \n", namespaceObject.OwnerReferences)
+			fmt.Printf("conditions %v \n", namespaceObject.Status.Conditions)
+
+			result, err := virtCli.RestClient().Get().RequestURI("/apis/apiregistration.k8s.io/v1/").Resource("apiservices").Do().Raw()
+			if err != nil {
+				fmt.Printf("could not get apiservices: %v\n", err)
+			}
+
+			apiServices := apiservices.APIServiceList{}
+			err = json.Unmarshal(result, &apiServices)
+			if err != nil {
+				fmt.Printf("could not unmarshal apiservices: %v\n", err)
+			}
+			for _, apiService := range apiServices.Items {
+				if apiService.Name == "v1alpha1.upload.cdi.kubevirt.io" {
+					fmt.Printf("apiservice '%s' '%s' status: \n %+v", apiService.Name, apiService.Spec.Group, apiService.Status.Conditions)
+				}
+				for _, condition := range apiService.Status.Conditions {
+					if condition.Type == "Available" {
+						if condition.Status == "False" {
+							fmt.Printf("apiservice '%s' '%s' not Available: \n%v\n", apiService.Name, apiService.Spec.Group, condition)
+						}
+					}
+				}
+			}
+
+			return errors.IsNotFound(namespaceDeletionError)
```

Comment:
consider returning an error and assert on that using Petr's cool Eventually.
like I commented here : 
https://github.com/kubevirt/ovs-cni/pull/124#discussion_r472140476

Reply:
Done